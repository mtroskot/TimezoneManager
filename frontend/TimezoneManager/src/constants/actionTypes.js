export const userActionTypes = Object.freeze({
  AUTH_AUTO_SIGN_IN: 'AUTH_AUTO_SIGN_IN',
  AUTH_USER: 'AUTH_USER',
  AUTH_USER_SUCCESS: 'AUTH_USER_SUCCESS',
  REGISTER_USER: 'REGISTER_USER',
  REGISTER_USER_SUCCESS: 'REGISTER_USER_SUCCESS',
  UPDATE_USER_INFO: 'UPDATE_USER_INFO',
  UPDATE_USER_INFO_SUCCESS: 'UPDATE_USER_INFO_SUCCESS',
  DELETE_USER: 'DELETE_USER',
  DELETE_USER_SUCCESS: 'DELETE_USER_SUCCESS',
  LOGOUT: 'LOGOUT',
  LOGOUT_SUCCESS: 'LOGOUT_SUCCESS',
  CHANGE_USER_ROLE: 'CHANGE_USER_ROLE',
  CHANGE_USER_ROLE_SUCCESS: 'CHANGE_USER_ROLE_SUCCESS'
});

export const uiActionTypes = Object.freeze({
  START_ACTION: 'START_ACTION',
  STOP_ACTION: 'STOP_ACTION',
  REFRESH_ACTION_START: 'REFRESH_ACTION_START',
  REFRESH_ACTION_STOP: 'REFRESH_ACTION_STOP',
  TOGGLE_POPUP_MESSAGE: 'TOGGLE_POPUP_MESSAGE',
  ADD_ERROR_ACTION: 'ADD_ERROR_ACTION',
  CLEAR_ERROR_ACTIONS: 'CLEAR_ERROR_ACTIONS'
});

export const timezoneActionTypes = Object.freeze({
  ADD_NEW_TIMEZONE_ENTRY: 'ADD_NEW_TIMEZONE_ENTRY',
  ADD_NEW_TIMEZONE_ENTRY_SUCCESS: 'ADD_NEW_TIMEZONE_ENTRY_SUCCESS',
  UPDATE_TIMEZONE_ENTRY: 'UPDATE_TIMEZONE_ENTRY',
  UPDATE_TIMEZONE_ENTRY_SUCCESS: 'UPDATE_TIMEZONE_ENTRY_SUCCESS',
  DELETE_TIMEZONE_ENTRY: 'DELETE_TIMEZONE_ENTRY',
  DELETE_TIMEZONE_ENTRY_SUCCESS: 'DELETE_TIMEZONE_ENTRY_SUCCESS',
  FETCH_TIMEZONE_ENTRIES: 'FETCH_TIMEZONE_ENTRIES',
  FETCH_TIMEZONE_ENTRIES_SUCCESS: 'FETCH_TIMEZONE_ENTRIES_SUCCESS'
});

export const searchActionTypes = Object.freeze({
  SEARCH_TIMEZONE_ENTRIES: 'SEARCH_TIMEZONE_ENTRIES',
  SEARCH_TIMEZONE_ENTRIES_SUCCESS: 'SEARCH_TIMEZONE_ENTRIES_SUCCESS',
  SEARCH_USERS: 'SEARCH_USERS',
  SEARCH_USERS_SUCCESS: 'SEARCH_USERS_SUCCESS',
  CLEAR_SEARCH: 'CLEAR_SEARCH'
});
