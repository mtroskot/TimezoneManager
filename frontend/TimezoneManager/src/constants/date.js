export const clockFormat = 'HH:mm:ss';
export const dateFormat = 'E, MMMM d';
