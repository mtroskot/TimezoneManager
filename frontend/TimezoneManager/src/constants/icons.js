export const icons = Object.freeze({
  GLOBE: 'globe',
  SETTINGS: 'settings'
});
