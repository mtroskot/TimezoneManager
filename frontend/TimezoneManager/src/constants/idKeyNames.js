export const idNames = Object.freeze({
  USER_ID: 'id',
  TIMEZONE_ENTRY_ID: 'id'
});
