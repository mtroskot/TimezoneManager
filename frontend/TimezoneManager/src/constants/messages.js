export const NO_INTERNET = 'No internet connection';
export const DEFAULT_ERROR = 'Something went wrong';
export const LOGIN_FAIL = 'Invalid email/password';
export const EMAIL_IN_USE = 'Email Address already in use';
export const NO_RESULTS = 'No Results Found';
export const REGISTRATION_SUCCESS = 'Registration Success';
