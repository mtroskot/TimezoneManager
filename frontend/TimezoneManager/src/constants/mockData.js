export const mockEntries = [
  {
    id: 1,
    name: 'Foo',
    city: 'Bar',
    differenceToGMT: -6
  },
  {
    id: 2,
    name: 'Foo',
    city: 'Bar',
    differenceToGMT: -6
  },
  {
    id: 3,
    name: 'Foo',
    city: 'Bar',
    differenceToGMT: -6
  },
  {
    id: 4,
    name: 'Foo',
    city: 'Bar',
    differenceToGMT: -6
  },
  {
    id: 5,
    name: 'Foo',
    city: 'Bar',
    differenceToGMT: -6
  },
  {
    id: 6,
    name: 'Foo',
    city: 'Bar',
    differenceToGMT: -6
  },
  {
    id: 7,
    name: 'Foo',
    city: 'Bar',
    differenceToGMT: -6
  },
  {
    id: 8,
    name: 'Foo',
    city: 'Bar',
    differenceToGMT: -6
  },
  {
    id: 9,
    name: 'Foo',
    city: 'Bar',
    differenceToGMT: -6
  }
];

export const mockUsers = [
  {
    id: 1,
    firstName: 'Marko',
    lastName: 'Troskot',
    emailAddress: 'marko.troskot@hotmail.commarko.troskot@hotmail.com'
  },
  { id: 2, firstName: 'Marko', lastName: 'Troskot', emailAddress: 'marko.troskot@hotmail.com' },
  { id: 3, firstName: 'Marko', lastName: 'Troskot', emailAddress: 'marko.troskot@hotmail.com' },
  { id: 4, firstName: 'Marko', lastName: 'Troskot', emailAddress: 'marko.troskot@hotmail.com' },
  { id: 5, firstName: 'Marko', lastName: 'Troskot', emailAddress: 'marko.troskot@hotmail.com' },
  { id: 6, firstName: 'Marko', lastName: 'Troskot', emailAddress: 'marko.troskot@hotmail.com' },
  { id: 7, firstName: 'Marko', lastName: 'Troskot', emailAddress: 'marko.troskot@hotmail.com' }
];
