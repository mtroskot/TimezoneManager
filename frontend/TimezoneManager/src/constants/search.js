export const filterOptions = Object.freeze({
  OWN_ENTRIES: 'OWN_ENTRIES',
  USERS: 'USERS',
  ALL_ENTRIES: 'ALL_ENTRIES'
});
