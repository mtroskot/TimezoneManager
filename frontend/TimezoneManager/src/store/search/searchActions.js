import { searchActionTypes } from 'src/constants/actionTypes';

export function searchTimezoneEntries(searchInput, cancelToken) {
  return {
    type: searchActionTypes.SEARCH_TIMEZONE_ENTRIES,
    payload: { searchInput, cancelToken }
  };
}

export const searchTimezoneEntriesSuccess = searchData => {
  return {
    type: searchActionTypes.SEARCH_TIMEZONE_ENTRIES_SUCCESS,
    payload: { searchData }
  };
};

export function searchAllTimezoneEntries(searchInput, cancelToken) {
  return {
    type: searchActionTypes.SEARCH_ALL_TIMEZONE_ENTRIES,
    payload: { searchInput, cancelToken }
  };
}

export const searchAllTimezoneEntriesSuccess = searchData => {
  return {
    type: searchActionTypes.SEARCH_ALL_TIMEZONE_ENTRIES_SUCCESS,
    payload: { searchData }
  };
};

export function searchUsers(searchInput, cancelToken) {
  return {
    type: searchActionTypes.SEARCH_USERS,
    payload: { searchInput, cancelToken }
  };
}

export const searchUsersSuccess = searchData => {
  return {
    type: searchActionTypes.SEARCH_USERS_SUCCESS,
    payload: { searchData }
  };
};

export function updateSearchedUserInfoSuccess(updatedUserInfo) {
  return {
    type: searchActionTypes.UPDATE_SEARCHED_USER_INFO_SUCCESS,
    payload: { updatedUserInfo }
  };
}

export const clearAllSearches = () => {
  return {
    type: searchActionTypes.CLEAR_ALL_SEARCHES,
    payload: {}
  };
};

export const clearTimezoneEntriesSearch = () => {
  return {
    type: searchActionTypes.CLEAR_TIMEZONE_ENTRIES_SEARCH,
    payload: {}
  };
};

export const clearAllTimezoneEntriesSearch = () => {
  return {
    type: searchActionTypes.CLEAR_ALL_TIMEZONE_ENTRIES_SEARCH,
    payload: {}
  };
};

export const clearUsersSearch = () => {
  return {
    type: searchActionTypes.CLEAR_USERS_SEARCH,
    payload: {}
  };
};
