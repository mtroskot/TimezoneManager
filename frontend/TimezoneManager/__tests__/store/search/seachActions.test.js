import configureStore from 'redux-mock-store';
import * as searchActions from 'src/store/search/searchActions';
import { searchActionTypes } from 'src/constants/actionTypes';

const middlewares = [];
const mockStore = configureStore(middlewares);
// Initialize mockstore with empty state
const initialState = {};
const store = mockStore(initialState);

describe('Search actions tests', () => {
  beforeEach(() => {
    store.clearActions();
  });

  it('dispatches searchTimezoneEntries action', () => {
    const searchInput = 'searchInput';
    const cancelToken = 'cancelToken';
    const fromAllUsers = 'fromAllUsers';
    const expectedActions = [
      {
        type: searchActionTypes.SEARCH_TIMEZONE_ENTRIES,
        payload: { searchInput, cancelToken, fromAllUsers }
      }
    ];
    store.dispatch(searchActions.searchTimezoneEntries(searchInput, cancelToken, fromAllUsers));
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });

  it('dispatches searchTimezoneEntriesSuccess action', () => {
    const searchData = 'searchData';
    const expectedActions = [
      {
        type: searchActionTypes.SEARCH_TIMEZONE_ENTRIES_SUCCESS,
        payload: { searchData }
      }
    ];
    store.dispatch(searchActions.searchTimezoneEntriesSuccess(searchData));
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });

  it('dispatches searchUsers action', () => {
    const searchInput = 'searchInput';
    const cancelToken = 'cancelToken';
    const expectedActions = [
      {
        type: searchActionTypes.SEARCH_USERS,
        payload: { searchInput, cancelToken }
      }
    ];
    store.dispatch(searchActions.searchUsers(searchInput, cancelToken));
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });

  it('dispatches searchUsersSuccess action', () => {
    const searchData = 'searchData';
    const expectedActions = [
      {
        type: searchActionTypes.SEARCH_USERS_SUCCESS,
        payload: { searchData }
      }
    ];
    store.dispatch(searchActions.searchUsersSuccess(searchData));
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });

  it('dispatches updateSearchedUserInfoSuccess action', () => {
    const updatedUserInfo = 'updatedUserInfo';
    const expectedActions = [
      {
        type: searchActionTypes.UPDATE_SEARCHED_USER_INFO_SUCCESS,
        payload: { updatedUserInfo }
      }
    ];
    store.dispatch(searchActions.updateSearchedUserInfoSuccess(updatedUserInfo));
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });

  it('dispatches clearAllSearches action', () => {
    const expectedActions = [
      {
        type: searchActionTypes.CLEAR_ALL_SEARCHES,
        payload: {}
      }
    ];
    store.dispatch(searchActions.clearAllSearches());
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });

  it('dispatches clearTimezoneEntriesSearch action', () => {
    const expectedActions = [
      {
        type: searchActionTypes.CLEAR_TIMEZONE_ENTRIES_SEARCH,
        payload: {}
      }
    ];
    store.dispatch(searchActions.clearTimezoneEntriesSearch());
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });

  it('dispatches clearUsersSearch action', () => {
    const expectedActions = [
      {
        type: searchActionTypes.CLEAR_USERS_SEARCH,
        payload: {}
      }
    ];
    store.dispatch(searchActions.clearUsersSearch());
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });
});
