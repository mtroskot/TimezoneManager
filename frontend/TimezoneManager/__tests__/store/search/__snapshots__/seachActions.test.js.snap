// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search actions tests dispatches clearAllSearches action 1`] = `
Array [
  Object {
    "payload": Object {},
    "type": "CLEAR_ALL_SEARCHES",
  },
]
`;

exports[`Search actions tests dispatches clearAllTimezoneEntriesSearch action 1`] = `
Array [
  Object {
    "payload": Object {},
    "type": "CLEAR_ALL_TIMEZONE_ENTRIES_SEARCH",
  },
]
`;

exports[`Search actions tests dispatches clearTimezoneEntriesSearch action 1`] = `
Array [
  Object {
    "payload": Object {},
    "type": "CLEAR_TIMEZONE_ENTRIES_SEARCH",
  },
]
`;

exports[`Search actions tests dispatches clearUsersSearch action 1`] = `
Array [
  Object {
    "payload": Object {},
    "type": "CLEAR_USERS_SEARCH",
  },
]
`;

exports[`Search actions tests dispatches searchAllTimezoneEntries action 1`] = `
Array [
  Object {
    "payload": Object {
      "cancelToken": "cancelToken",
      "searchInput": "searchInput",
    },
    "type": "SEARCH_ALL_TIMEZONE_ENTRIES",
  },
]
`;

exports[`Search actions tests dispatches searchAllTimezoneEntriesSuccess action 1`] = `
Array [
  Object {
    "payload": Object {
      "searchData": "searchData",
    },
    "type": "SEARCH_ALL_TIMEZONE_ENTRIES_SUCCESS",
  },
]
`;

exports[`Search actions tests dispatches searchTimezoneEntries action 1`] = `
Array [
  Object {
    "payload": Object {
      "cancelToken": "cancelToken",
      "searchInput": "searchInput",
    },
    "type": "SEARCH_TIMEZONE_ENTRIES",
  },
]
`;

exports[`Search actions tests dispatches searchTimezoneEntriesSuccess action 1`] = `
Array [
  Object {
    "payload": Object {
      "searchData": "searchData",
    },
    "type": "SEARCH_TIMEZONE_ENTRIES_SUCCESS",
  },
]
`;

exports[`Search actions tests dispatches searchUsers action 1`] = `
Array [
  Object {
    "payload": Object {
      "cancelToken": "cancelToken",
      "searchInput": "searchInput",
    },
    "type": "SEARCH_USERS",
  },
]
`;

exports[`Search actions tests dispatches searchUsersSuccess action 1`] = `
Array [
  Object {
    "payload": Object {
      "searchData": "searchData",
    },
    "type": "SEARCH_USERS_SUCCESS",
  },
]
`;

exports[`Search actions tests dispatches updateSearchedUserInfoSuccess action 1`] = `
Array [
  Object {
    "payload": Object {
      "updatedUserInfo": "updatedUserInfo",
    },
    "type": "UPDATE_SEARCHED_USER_INFO_SUCCESS",
  },
]
`;
